### PR TITLE
APIKIT-2735: APIKit handling for out of scope OAS 3 features

### PR DESCRIPTION
**Pull request by dcourtil**

### CODE REVIEW

File: src/main/java/org/mule/module/apikit/api/RamlHandler.java [hunk 1]
```diff
@@ -112,6 +114,14 @@ public RamlHandler(ScheduledExecutorService executor,
     }
   }
 
+  private void handleUnsupportedFeaturesLogging(ApiReference apiReference) {
+    List<ParsingIssue> warnings = result.getWarnings();
+    if (ApiVendor.OAS_30.equals(apiReference.getVendor()) && isNotEmpty(warnings)) {
+      String warnsMessage = warnings.stream().map(e -> "  - " + e.cause()).collect(Collectors.joining(" \n"));
+      LOGGER.warn(getBoilerPlate("WARNING: OAS 3.0 is not fully supported yet\n" + warnsMessage));
```
[the remark this answers is not in the document]
Comment (dcourtil) [reply src/main/java/org/mule/module/apikit/api/RamlHandler.java:121]: Yeah, the message that OAS 3 is not fully supported might bring confusion if the warning has nothing to do with callbacks and links but I don't consider it harmful. Not sure how frequently a warning is sent back to us by AMF but let's discuss a different approach for this.

[the remark this answers is not in the document]
Comment (dcourtil) [reply src/main/java/org/mule/module/apikit/api/RamlHandler.java]: As discussed offline, updated to log any warning coming from the parser.

File: src/main/java/org/mule/module/apikit/api/RamlHandler.java [hunk 2]
```diff
@@ -112,6 +114,13 @@ public RamlHandler(ScheduledExecutorService executor,
     }
   }
 
+  private void handleWarnings(ApiReference apiReference) {
+    List<ParsingIssue> warnings = result.getWarnings();
+    if (ApiVendor.OAS_30.equals(apiReference.getVendor()) && isNotEmpty(warnings)) {
```
Comment (juanchib) [inline src/main/java/org/mule/module/apikit/api/RamlHandler.java:119]: Shouldn't we log all warnings ? just a question to review before merging this

Comment (dcourtil) [reply src/main/java/org/mule/module/apikit/api/RamlHandler.java:119]: Good point! I think that doing so will be more consistent

Comment (dcourtil) [reply src/main/java/org/mule/module/apikit/api/RamlHandler.java]: Updated! All warnings will be logged if any. 
Anyway, it is always possible to update the logging level of apikit by updating the the log4j2.xml of the app.
`<AsyncLogger name="org.mule.module.apikit" level="ERROR"/>`

File: src/main/java/org/mule/module/apikit/api/RamlHandler.java [hunk 3]
```diff
@@ -93,9 +94,10 @@ public RamlHandler(ScheduledExecutorService executor,
     if (rootRamlLocation == null) {
       throw new IOException("Raml not found at: " + ramlLocation);
     }
-
-    result = parserService.parse(ApiReference.create(rootRamlLocation), parserMode == null ? AUTO : parserMode);
+    ApiReference apiReference = ApiReference.create(rootRamlLocation);
+    result = parserService.parse(apiReference, parserMode == null ? AUTO : parserMode);
     if (result.success()) {
+      handleWarnings();
```
Comment (juanchib) [inline src/main/java/org/mule/module/apikit/api/RamlHandler.java:100]: logWarnings (?)

File: src/main/java/org/mule/module/apikit/api/RamlHandler.java [hunk 4]
```diff
@@ -107,11 +109,18 @@ public RamlHandler(ScheduledExecutorService executor,
...
       throw new RuntimeException("Errors while parsing RAML file in [" + parserMode + "] mode: \n" + errors);
     }
   }
 
+  private void handleWarnings() {
+    List<ParsingIssue> warnings = result.getWarnings();
+    if (isNotEmpty(warnings)) {
+      LOGGER.warn(getBoilerPlate(warnings.stream().map(e -> "  - " + e.cause()).collect(joining(" \n"))));
+    }
+  }
```
Comment (juanchib) [inline src/main/java/org/mule/module/apikit/api/RamlHandler.java:122]: I would make it static and receive result.getWarnings() as parameter

Comment (juanchib) [reply src/main/java/org/mule/module/apikit/api/RamlHandler.java:122]: you can't access LOGGER either so disregard this comment

File: src/main/java/org/mule/module/apikit/api/RamlHandler.java (hunk 3 above)
Comment (dcourtil) [reply src/main/java/org/mule/module/apikit/api/RamlHandler.java]: Updated

**Review by juanchib — APPROVED** [review]

**Pull request merged**